src/dss/create_notebook.py

```diff
@@ -0,0 +1,129 @@
+from pathlib import Path
+
+from charmed_kubeflow_chisme.kubernetes import KubernetesResourceHandler
+from lightkube import Client
+from lightkube.core.exceptions import ApiError
+from lightkube.resources.apps_v1 import Deployment
+from lightkube.resources.core_v1 import Service
+
+from dss.config import (
+    DSS_CLI_MANAGER_LABELS,
+    DSS_NAMESPACE,
+    FIELD_MANAGER,
+    MANIFEST_TEMPLATES_LOCATION,
+    NOTEBOOK_IMAGES_ALIASES,
+    NOTEBOOK_PVC_NAME,
+    RECOMMENDED_IMAGES_MESSAGE,
+)
+from dss.logger import setup_logger
+from dss.utils import (
+    ImagePullBackOffError,
+    does_dss_pvc_exist,
+    does_mlflow_deployment_exist,
+    does_notebook_exist,
+    get_mlflow_tracking_uri,
+    get_notebook_url,
+    wait_for_deployment_ready,
+)
+
+# Set up logger
+logger = setup_logger("logs/dss.log")
+
+
+def create_notebook(name: str, image: str, lightkube_client: Client) -> None:
+    """
+    Creates a Notebook server on the Kubernetes cluster.
+
+    Args:
+        name (str): The name of the notebook server.
+        image (str): The image used for the notebook server.
+        lightkube_client (Client): The Kubernetes client.
+    """
+    if not does_dss_pvc_exist(lightkube_client) or not does_mlflow_deployment_exist(
+        lightkube_client
+    ):
+        logger.error(
```

Comment:
We are little inconsistent with these messages .... in logs [here](https://github.com/canonical/data-science-stack/blob/main/src/dss/logs.py#L50) we write the first line as an error log while we write the hints as infos. Here we put everything in error log. The reasoning behind logs design was that info logs and error logs are stored in separate files for dss. With this in mind we are putting hints into info logs as when sb will we checking error logs they dont need the hints there.

Reply:
good one, updated!